### ISSUE TITLE
Link as a slide

### ISSUE THREAD

**Issue by care4pet** [maxmarinich/react-alice-carousel#144]
Hello, thanks again for great work and progress!

There is an issue when you use link as a slide.
The first problem is that default Chrome behaviour is dragging link, instead of passing it to the library. 

It can be easily fixed by 
```
a {
   -webkit-user-drag: none;
   user-select: none;
}
```

in `alice-carousel.scss`

The less trivial problem is that it still passing events to the link when user release mouse. 
So user slides by mouse, release it and click event processes. 

I think it would be great to prevent it, but keep if no movement happened and user just clicked on a slide.

To reproduce it, just replace divs with links like that 
```
<AliceCarousel
	mouseTracking
	responsive={{
		0: { items: 1 },
		568: { items: 2 },
		1024: { items: 3 },
	}}
>
	<a className="item" href="#">
		<h1>1</h1>
	</a>
	<a className="item" href="#">
		<h1>2</h1>
	</a>
	<a className="item" href="#">
		<h1>3</h1>
	</a>
	<a className="item" href="#">
		<h1>4</h1>
	</a>
	<a className="item" href="#">
		<h1>5</h1>
	</a>
</AliceCarousel>
```

**Comment by care4pet** [comment]
I tried to fix it by myself, but I didn't find strainforward solution for that.

**Comment by Saredon** [comment]
Hi, @care4pet.
We wrapped child elements with some functions to prevent default click event when elements are dragged.

```
let isDragging = false;
const handleOnDragStart = (e) => {
	e.preventDefault();
	isDragging = true;
};

const onClickCapture = (e) => {
	if (isDragging) {
		isDragging = false;
		e.preventDefault();
		return;
	}
};

const items = Children.map(children, (child) => {
	return (<div onDragStart={handleOnDragStart} onDrag={handleOnDragStart} draggable
			onClickCapture={onClickCapture}>{child}</div>);
});

return (<AliceCarousel animationDuration={250} items={items} touchMoveDefaultEvents mouseTracking {...props} />)
```

It's not necessary to wrap them in <div>. You can clone child elements and pass these functions to them.
We created the component to prevent click event in all places where we use this library.

I hope it helps you.

**Comment by maxmarinich** [comment]
Hi, @care4pet! Like @Saredon have noticed, you should prevent default `drag` event:
```js
import React from 'react';
import AliceCarousel from 'react-alice-carousel';
import 'react-alice-carousel/lib/alice-carousel.css';


const handleDragStart = (e) => e.preventDefault();

const items = [
  <a href="#" onDragStart={handleDragStart} className="yours-custom-class" />,
  <a href="#" onDragStart={handleDragStart} className="yours-custom-class" />,
  <a href="#" onDragStart={handleDragStart} className="yours-custom-class" />,
];

const Gallery = () => {
  return (
    <AliceCarousel mouseTracking items={items} />
  );
}
```

**Comment by TheHelias** [comment]
Still experiencing issues with this. Items that are links get clicked after I am done dragging

**Comment by websuits** [comment]
@TheHelias check @Saredon 's solution. That one will work. Max's solution is incomplete. Cheers !

**Comment by scottroot** [comment]
@Saredon thank you.  After wasting a lot of time trying different attributes and hooks, your solution works perfectly.

**Comment by stevensturkop** [comment]
@saredon 's solution does not work for me. Every time I drag the carousel, despite the handleDragStart, it clicks on the link when I release the mouse.

```
import Image from 'next/image';
import React from 'react';
import AliceCarousel from 'react-alice-carousel';
import 'react-alice-carousel/lib/alice-carousel.css';

const handleDragStart = (e: React.DragEvent<HTMLAnchorElement>) => e.preventDefault();

const items = [
  <a href='#' key='1' onDragStart={handleDragStart}>
    <Image src='/assets/images/banner-alcohol.png' alt='test' width={600} height={400} />
  </a>,
  <a href='#' key='2' onDragStart={handleDragStart}>
    <Image src='/assets/images/banner-alcohol.png' alt='test' width={600} height={400} />
  </a>,
  <a href='#' key='3' onDragStart={handleDragStart}>
    <Image src='/assets/images/banner-alcohol.png' alt='test' width={600} height={400} />
  </a>,
];

export const Gallery = () => {
  return <AliceCarousel mouseTracking items={items} />;
};
```

**Comment by maxmarinich** [comment]
Resolved at #314 . Use <Link /> component like [here](https://maxmarinich.github.io/react-alice-carousel/#custom-components-links).
